src/overlays/actors/ovl_En_Heishi1/z_en_heishi1.c

```diff
@@ -21,37 +43,473 @@ const ActorInit En_Heishi1_InitVars = {
+    Camera* activeCam;
+    Vec3f searchBallVel;
+    Vec3f searchBallAccel;
+    Vec3f searchBallMult;
+    Vec3f searchBallPos;
+
+    this->activeTimer++;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi1/func_80A5162C.s")
+    for (i = 0; i < ARRAY_COUNT(this->timers); i++) {
+        if (this->timers[i] != 0) {
+            this->timers[i]--;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi1/func_80A516E4.s")
+    if (this->waypointTimer != 0) {
+        this->waypointTimer--;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi1/func_80A51A98.s")
+    activeCam = ACTIVE_CAM;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi1/func_80A51B54.s")
+    if (player->actor.freezeTimer == 0) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi1/func_80A51C4C.s")
+        this->actionFunc(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi1/func_80A51D18.s")
+        this->actor.uncullZoneForward = 550.0f;
+        this->actor.uncullZoneScale = 350.0f;
+        this->actor.uncullZoneDownward = 700.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi1/func_80A51F50.s")
+        if (this->type != 5) {
+            path = this->path * 2;
+            if ((sCamDataIdxs[path] == activeCam->unk_148) || (sCamDataIdxs[path + 1] == activeCam->unk_148)) {
+                if (!sPlayerIsCaught) {
+                    if ((this->actionFunc == EnHeishi1_Walk) || (this->actionFunc == EnHeishi1_Wait)) {
+                        searchBallAccel = sSearchBallAccel;
+                        searchBallMult = sSearchBallMult;
+                        searchBallPos.x = this->actor.posRot.pos.x;
+                        searchBallPos.y = this->actor.posRot.pos.y + 60.0f;
+                        searchBallPos.z = this->actor.posRot.pos.z;
+                        Matrix_Push();
+                        Matrix_RotateY(((this->actor.shape.rot.y + this->headAngle) / 32768.0f) * M_PI, 0);
+                        searchBallMult.z = 30.0f;
+                        Matrix_MultVec3f(&searchBallMult, &searchBallVel);
+                        Matrix_Pull();
+                        EffSsSolderSrchBall_Spawn(globalCtx, &searchBallPos, &searchBallVel, &searchBallAccel, 2,
+                                                  &this->linkDetected);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Heishi1/func_80A51FEC.s")
+                        if (this->actor.xzDistFromLink < 60.0f) {
+                            this->linkDetected = true;
+                        } else if (this->actor.xzDistFromLink < 70.0f) {
+                            // this case probably exists to detect link making a sidhop or backflip sound
+                            // from slightly further away than the previous 60 unit check
+                            if (player->actor.velocity.y > -4.0f) {
```

Comment:
Doesn't this check also catch normal climbs/jumps? Imo this check is just designed to detect jumps in general, no need to try and guess which types are relevant exactly based on the paths and all.

Reply:
the only other place you can jump is the steps, and its so far away from a guard that it doesnt make sense for that. sidehops and backflips should be the only ones that matter.

Reply:
ill change it anyway just in case, but i really do think this is specifically for sidehops and backflips